src/test/java/io/cdap/plugin/splunk/common/client/SplunkSearchIteratorTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.splunk.common.client;
+
+import com.google.common.collect.ImmutableMap;
+import com.splunk.Event;
+import com.splunk.ResultsReader;
+import io.cdap.plugin.splunk.source.batch.SplunkBatchSourceConfig;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Tests for {@link SplunkSearchIterator}
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(ResultsReader.class)
+@PowerMockIgnore({"javax.net.ssl.*"})
+public class SplunkSearchIteratorTest {
+
+  private static final SplunkBatchSourceConfig CONFIG = new SplunkBatchSourceConfig(
+    "reference",
+    "basic",
+    "apiToken",
+    "userName",
+    60000,
+    60000,
+    3,
+    60000,
+    100,
+    "invalid",
+    "password",
+    "executionMode",
+    "outputFormat",
+    "searchString",
+    "searchId",
+    0L,
+    "earliestTime",
+    "latestTime",
+    "indexedEarliestTime",
+    "indexedLatestTime",
+    100L,
+    "schema");
+  private static final String SEARCH_ID = "searchId";
+  private static final long OFFSET = 0L;
+
+  @Test
+  public void testSearchIterator() {
+    Map<String, String> expectedEvent1 = ImmutableMap.of("key1", "event1");
+
+    Map<String, String> event1 = ImmutableMap.of("!k^e*y()1{}", "event1");
+    Map<String, String> event2 = ImmutableMap.of("key1", "event2");
+    Map<String, String> event3 = ImmutableMap.of("key1", "event3");
+    List<Event> events = new ArrayList<>();
+    addEvent(events, event1);
+    addEvent(events, event2);
+    addEvent(events, event3);
+
+    InputStream stream = Mockito.mock(InputStream.class);
+    ResultsReader resultsReader = PowerMockito.mock(ResultsReader.class);
```

Comment:
why do you need PowerMockito here?

Reply:
com.splunk.ResultsReader.iterator() is final. Mockito has support for final methods but it is **incubating** feature.